foundation-extensions/mobile-core/lifecycle/README.md

```diff
@@ -136,7 +154,142 @@ using Com.Adobe.Marketing.Mobile;
+
+   If your iOS application supports background capabilities, your `application:didFinishLaunchingWithOptions:` method might be called when iOS launches your app in the background. If you do not want background launches to count towards your lifecycle metrics, then `lifecycleStart:` should only be called when the application state is not equal to `UIApplicationStateBackground`.
+
+   ```objectivec
+   - (BOOL) application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+       // register the lifecycle extension, and begin event processing with Core
+       const UIApplicationState appState = application.applicationState;
+      [AEPMobileCore registerExtensions:@[AEPMobileLifecycle.class, ...] completion:^{
+        // only start lifecycle if the application is not in the background
+        if (appState != UIApplicationStateBackground) {
+            [AEPMobileCore lifecycleStart:nil];
+        }
+      }];
+   }
+   ```
+
+3. When launched, if your app is resuming from a backgrounded state, iOS might call your `applicationWillEnterForeground:` delegate method. You also need to call `lifecycleStart:`, but this time you do not need all of the supporting code that you used in `application:didFinishLaunchingWithOptions:`:
+
+   ```objectivec
+   - (void) applicationWillEnterForeground:(UIApplication *)application {
+       [AEPMobileCore lifecycleStart:nil];
+   }
+   ```
+
+   In iOS 13 and later, for a scene-based application, use the `UISceneDelegate`'s `sceneWillEnterForeground` method as follows:
+
+   ```objectivec
+   - (void) sceneWillEnterForeground:(UIScene *)scene {
+      [AEPMobileCore lifecycleStart:nil];
+   }
+   ```
+
+   For more information on handling foregrounding applications with Scenes, refer to Apple's documentation [here](https://developer.apple.com/documentation/uikit/app_and_environment/scenes/preparing_your_ui_to_run_in_the_foreground)
+
+4. When the app enters the background, pause Lifecycle data collection from your app's `applicationDidEnterBackground:` delegate method:
+
+   ```objectivec
+    - (void) applicationDidEnterBackground:(UIApplication *)application {
+       [AEPMobileCore lifecyclePause];
+    }
+   ```
+
+   In iOS 13 and later, for a scene-based application, use the `UISceneDelegate`'s `sceneDidEnterBackground` method as follows:
+
+   ```objectivec
+   - (void) sceneDidEnterBackground:(UIScene *)scene {
+      [AEPMobileCore lifecyclePause];
+   }
+   ```
+
+   For more information on handling backgrounding applications with Scenes, refer to Apple's documentation [here](https://developer.apple.com/documentation/uikit/app_and_environment/scenes/preparing_your_ui_to_run_in_the_background)
+
+{% endtab %}
+
+{% tab title="iOS (ACP 2.x)" %}
+#### Objective-C
```

Comment:
A general comment: It would be great to have syntax + examples for Swift, followed by Objective-C across all pages and tabs. Should we consider making this change in a separate PR?

Reply:
I can add syntax + examples to the readme pages in this PR

Reply:
What do you think is the best way to do this for the Lifecycle readme? Since the format for lifecycle readme is unique in that we have a numbered step by step tutorial-like documentation?

Reply:
Even for Lifecycle we still need two tabs AEP (3.x) and ACP (2.x) for the iOS examples.
Few other improvements we can do:
- separate the registration of the extension from the start/pause implementation, so we have two sections: "Register Lifecycle with Mobile Core", followed by  "Implement Start/Pause APIs"
- Inside the iOS tabs for "Implement Start/Pause APIs", have the following format: 
    - Swift step-by-step tutorial
    - Objective-C step-by-step tutorial
    - In iOS 13 and later, for a scene-based application, use the UISceneDelegate's sceneWillEnterForeground and sceneDidEnterBackground method. For more information on handling foregrounding applications with Scenes, refer to Apple's documentation here.

Reply:
I think restructuring all of these docs might need to be done in another PR actually. Seems like a lot for this. This was mainly to add missing swift docs.

Reply:
👍  that sounds ok to me